cmake/FindZ3.cmake

```diff
@@ -2,3 +2,24 @@ find_path(Z3_INCLUDE_DIR NAMES z3.h)
 find_library(Z3_LIBRARIES NAMES z3)
```

Comment:
Then please fix this to specify the required version.

Reply:
ok, added version checking to cmake.

Reply:
Thank you.